### PR TITLE
Add CI workflows to test builds

### PR DESCRIPTION
**Pull request by alerque**
Follow up to linter job, not (yet) automated deploys this is just build checks

(Based on #355 so I don't have to redo rockspec location stuff, should not be merged first.)

### CODE REVIEW

**Comment by alerque** [comment]
This is not parity with what the obsolete Travis CI config was doing — and beyond since it is test building across 3 more Lua versions and macos in addition to the limited selection of aging Luas on Linux in the Travis matrix.

@Tieske I still have this in draft mode because I want to sort out and merge that rockspec related PR before this one, but this is ready to go as soon as you take a gander at that.

**Comment by alerque** [comment]
Also note this doesn't do any of the automated release stuff yet, just runs the regression test suite on various platforms.

**Review by daurnimator — COMMENTED** [review]
> chore: Rename rockspec dir to be plural 

Why?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          # lua testsupport.lua
+          # lua tftptest.lua
+          # lua udpconnectclnt.lua
+          # lua udpconnectsrvr.lua
+          # lua unixdgramclnt.lua
+          # lua unixdgramsrvr.lua
+          # lua unixstreamclnt.lua
+          # lua unixstreamsrvr.lua
+          # lua utestclnt.lua
+          # lua utestsrvr.lua
```
Comment (daurnimator) [inline .github/workflows/build.yml:60]: Why are these here and commented out?

Comment (alerque) [reply .github/workflows/build.yml:60]: Because I was playing around with files that were in tests but not part of the current suite. I'll remove them before taking this out of draft mode.

**Comment by Tieske** [comment]
> > chore: Rename rockspec dir to be plural
> 
> Why?

consistency. @alerque did some great work on automating releases upon tagging, but this does require some consitency (at least that's what I assume is the reason)

**Review by Tieske — APPROVED** [review]

**Comment by alerque** [comment]
Yes, what @Tieske said. It is a lot easier to reuse stuff like that when projects are consistent about stuff like that. And as long as I'm directly involved in the release process I'd rather normalize the project than add exceptions and variants to all the related tooling. Obviously both could be done, but given that grammatically a plural or collective noun is called for anyway fixing that seems like a better option all around.
